src/main/java/com/scalar/db/api/TableMetadata.java

```diff
@@ -0,0 +1,172 @@
+package com.scalar.db.api;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.scalar.db.api.Scan.Ordering.Order;
+import com.scalar.db.io.DataType;
+import com.scalar.db.storage.common.util.ImmutableLinkedHashSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/** Represents the table metadata */
```

Comment:
It's not in the google style guide, but I think it would be a bit better and consistent if a comment for a class starts with a noun.